Support PKShippingMethods on iOS

JonWBedard opened the pull request:
#### fe9ec83a70d003c1ebd159c4cc8fc268e4a6787e
<pre>
Support PKShippingMethods on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=238845">https://bugs.webkit.org/show_bug.cgi?id=238845</a>
&lt;rdar://problem/91320467 &gt;

Reviewed by Devin Rousso.

* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h: Move PKShippingMethods SPI declarations out of MacOS specific #ifdefs.

Canonical link: <a href="https://commits.webkit.org/249321@main">https://commits.webkit.org/249321@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@292470">https://svn.webkit.org/repository/webkit/trunk@292470</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

dcrousso commented on Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
```diff
@@ -549,6 +536,24 @@ NS_ASSUME_NONNULL_END
+@interface PKPaymentRequestUpdate ()
+@property (nonatomic, copy) PKShippingMethods *availableShippingMethods;
+@end
+#endif
```
NIT: I'd add a `// HAVE(PASSKIT_DEFAULT_SHIPPING_METHOD) && !USE(APPLE_INTERNAL_SDK)` here so that it's clear what this is ending.

JonWBedard committed:
Support PKShippingMethods on iOS
https://bugs.webkit.org/show_bug.cgi?id=238845
<rdar://problem/91320467>

Patch by Jonathan Bedard <jbedard@apple.com> on 2022-04-06
Reviewed by Devin Rousso.

* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h: Move PKShippingMethods SPI declarations out of MacOS specific #ifdefs.
